service.go

```diff
@@ -151,6 +152,85 @@ func setupTempLog() *Logs {
 	return &logs
 }
 
+func SwitchLogLevel(logfile string, minLogLevel string) (LogType, error) {
+	// validate log level
+	minLogLevelType, err := validateLogType(minLogLevel)
+	if err != nil {
+		return minLogLevelType, err
+	}
+
+	if Log == nil {
+		//  initialize logger
+		SetupLogWriters(logfile, []io.Writer{}, minLogLevelType)
+		return minLogLevelType, nil
+	}
+
+	var levelWriter = Log.Error.Writer()
+
+	switch minLogLevelType {
+	case Trace:
+		setLoggers(Trace, levelWriter)
+	case Debug:
+		Log.Trace.SetOutput(io.Discard)
+		setLoggers(Debug, levelWriter)
+	case Info:
+		Log.Trace.SetOutput(io.Discard)
+		Log.Debug.SetOutput(io.Discard)
+		setLoggers(Info, levelWriter)
+	case Warning:
+		Log.Trace.SetOutput(io.Discard)
+		Log.Debug.SetOutput(io.Discard)
+		Log.Info.SetOutput(io.Discard)
+		setLoggers(Warning, levelWriter)
+	case Error:
+		Log.Trace.SetOutput(io.Discard)
+		Log.Debug.SetOutput(io.Discard)
+		Log.Info.SetOutput(io.Discard)
+		Log.Warning.SetOutput(io.Discard)
```

Comment:
Do we not always have a LogLevel of Error or do we want to be able to remove all levels?
Current functionality is that if **Error** is sent then we will discard all lower levels but keep **Error**.

Reply:
Yeah correct, but just not sure why this particular `case` didn't have a `setLoggers` call like the others. Perhaps it was unnecessary. But even so it's probably a moot point if you're simplifying it as this would all be removed.

Reply:
`Error` logger is never set to `io.Discard` so never needs to be set again. But yeah refactoring the logic will remove this.

Reply:
Ah-ha I see. I guess when refactoring you can iterate the levels and either create or `SetOutput`... or possibly simpler, just recreate it every time (they won't get used very often!).

One other thing I noticed in the existing code. Perhaps at the bottom of `SetupLogWriters` we should be assigning the global `Logs` var? It seems the global one is only assigned when setting up the default logs. Perhaps I'm missing something there.

Reply:
Removed `setLoggers` function and the switch statement from `SwitchLogLevel` which now calls 	`setLogLevel(minLogLevelType)` instead.